apps/passport-client/src/service-worker.ts

```diff
@@ -1,85 +1,307 @@
+        // cache, if possible.
+        const cacheResp = await fetchFromCache(
+          EPHEMERAL_CACHE_NAME,
+          event.request
+        );
+        if (cacheResp) {
+          swLog.V(`cache fallback ${event.request?.url}`);
+          return cacheResp;
+        }
+
+        // If we have no cached entry available, fail with the fetch's error.
+        throw error;
+      }
+    })()
+  );
+});
+
+/**
+ * Cache setup which occurs during installation.
+ */
+async function prePopulateCaches(): Promise<void> {
+  await addResourcesToCache(STABLE_CACHE_NAME, STABLE_CACHE_RESOURCES);
+  await addResourcesToCache(EPHEMERAL_CACHE_NAME, EPHEMERAL_CACHE_RESOURCES);
+}
 
-  const keys = await cache.keys();
+/**
+ * Cache cleanup which occurs on activation.
+ */
+async function deleteOldCaches(): Promise<void> {
+  // Delete old caches that don't match the current version.
+  for (const cacheName of await caches.keys()) {
+    if (cacheName !== STABLE_CACHE_NAME && cacheName !== EPHEMERAL_CACHE_NAME) {
+      swLog.I(`Deleting unknown cache version: ${cacheName}`);
+      await caches.delete(cacheName);
+    }
+  }
 
   // Delete all existing cache entries not in our "permanent" list.
   // This eliminates any stale entries from old service workers which
   // used a different list.
+  const stableCache = await caches.open(STABLE_CACHE_NAME);
+  const keys = await stableCache.keys();
   await Promise.all(
     keys.map(async (request: Request) => {
       const urlKey = requestToItemCacheKey(request);
-      if (!permanentCache.includes(urlKey)) {
-        console.log(`[SERVICE_WORKER] discarding ${urlKey}`);
-        await cache.delete(request);
+      if (!STABLE_CACHE_RESOURCES.includes(urlKey)) {
+        swLog.I(`${STABLE_CACHE_NAME} discarding ${urlKey}`);
+        await stableCache.delete(request);
       } else {
-        console.log(`[SERVICE_WORKER] keeping ${urlKey}`);
+        swLog.V(`${STABLE_CACHE_NAME} keeping ${urlKey}`);
       }
     })
   );
-
-  // Pre-populate the cache with the entries we want.
-  await cache.addAll([...impermanentCache, ...permanentCache]);
 }
 
-self.addEventListener("install", (event: any) => {
-  console.log(`[SERVICE_WORKER] installing ${process.env.SW_ID}`);
-  (self as any).skipWaiting();
+async function addResourcesToCache(
+  cacheName: string,
+  resources: string[]
+): Promise<void> {
+  const cache = await caches.open(cacheName);
+  for (const resource of resources) {
+    try {
+      await cache.add(resource);
+      swLog.V(`Cache ${logCacheName(cacheName)} pre-cached ${resource}`);
+    } catch (error) {
+      swLog.V(
+        `Cache ${logCacheName(
+          cacheName
+        )} failed to pre-cache ${resource}: ${error}`
+      );
+    }
+  }
+}
 
-  event.waitUntil(addResourcesToCache());
+function shouldHandleFetch(request: Request): boolean {
+  return (
+    SERVICE_WORKER_ENABLED &&
+    request.method === "GET" &&
+    swSelf.location.origin === new URL(request.url).origin
```

Comment:
just checking my understanding but your intention here is to make sure that the only things that are cached are resources served from the static website, not the server nor any other endpoint?